pkg/apis/runtime/v1alpha1/function_types.go

```diff
@@ -45,15 +45,35 @@ type FunctionSpec struct {
 	Env []v1.EnvVar `json:"env,omitempty"`
 }
 
+// TemplateKind defines the type of BuildTemplate used by the build.
+type FunctionCondition string
+
+const (
+	// Indicates that function has a running condition.
```

Comment:
What about having something like "unknown" or "N/A" status by default? Currently, then you create a function it is "" which feels like the status is broken. So use "unknown" or "N/A" as default status

Reply:
"unknown" it is a good one